### PR TITLE
Kotlin migration CardBrowserMySearchesDialog

### PR DESCRIPTION
**Pull request by Arnold2381**
Kotlin migration CardBrowserMySearchesDialog

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt [hunk 1]
```diff
@@ -1,126 +1,115 @@
...
+import com.afollestad.materialdialogs.DialogAction
+import com.afollestad.materialdialogs.MaterialDialog
+import com.ichi2.anki.R
+import com.ichi2.anki.analytics.AnalyticsDialogFragment
+import com.ichi2.ui.ButtonItemAdapter
+import timber.log.Timber
+import java.util.*
+
+class CardBrowserMySearchesDialog : AnalyticsDialogFragment() {
+    private var mButtonItemAdapter: ButtonItemAdapter? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt:18]: Maybe we can make these `lateinit`

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt [hunk 2]
```diff
@@ -1,126 +1,115 @@
...
-
-        Activity activity = getActivity();
-        final MaterialDialog.Builder builder = new MaterialDialog.Builder(activity);
-
-        int type = getArguments().getInt("type");
+    @Suppress("UNCHECKED_CAST")
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        super.onCreate(savedInstanceState)
+        val res = resources
+        val activity: Activity? = activity
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt:33]: followup: `requireActivity()`

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt [hunk 3]
```diff
@@ -1,126 +1,115 @@
...
-                        getDialog().dismiss();
-                    },
-                    searchName -> {
-                        Timber.d("button clicked: %s", searchName);
-                        removeSearch(searchName);
-                    });
-
+            mSavedFilters = requireArguments().getSerializable("savedFilters") as HashMap<String, String>?
+            mSavedFilterKeys = ArrayList(mSavedFilters!!.keys)
+            mButtonItemAdapter = ButtonItemAdapter(mSavedFilterKeys)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt:39]: followup: kotlin extension

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt [hunk 4]
```diff
@@ -1,126 +1,115 @@
...
-            LinearLayoutManager layoutManager = (LinearLayoutManager)dialog.getRecyclerView().getLayoutManager();
-            DividerItemDecoration dividerItemDecoration =
-                    new DividerItemDecoration(dialog.getRecyclerView().getContext(), layoutManager.getOrientation());
-            float scale = res.getDisplayMetrics().density;
-            int dpAsPixels = (int) (5*scale + 0.5f);
-            dialog.getView().setPadding(dpAsPixels, 0, dpAsPixels, dpAsPixels);
-            dialog.getRecyclerView().addItemDecoration(dividerItemDecoration);
+        val dialog = builder.build()
+        if (dialog.recyclerView != null) {
+            val layoutManager = dialog.recyclerView.layoutManager as LinearLayoutManager?
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt:65]: followup: nonnull

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt [hunk 5]
```diff
@@ -1,126 +1,115 @@
...
+            .negativeText(R.string.dialog_cancel)
+            .onPositive { dialog: MaterialDialog, _: DialogAction? ->
+                mMySearchesDialogListener!!.onRemoveSearch(searchName)
+                mSavedFilters!!.remove(searchName)
+                mSavedFilterKeys!!.remove(searchName)
+                mButtonItemAdapter!!.remove(searchName)
+                mButtonItemAdapter!!.notifyAdapterDataSetChanged()
+                dialog.dismiss()
+                if (mSavedFilters!!.isEmpty()) {
+                    getDialog()!!.dismiss()
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CardBrowserMySearchesDialog.kt:89]: `requireDialog()` I think

**Review by david-allison-1 — APPROVED** [review]
LGTM, few followups here

**Review by mikehardy — APPROVED** [review]
"what David said" haha

**Pull request merged**